src/utils/arrays.ts

```diff
@@ -45,3 +45,63 @@ export function arrayDiff<T>(a: T[], b: T[]): { added: T[], removed: T[] } {
         removed: a.filter(i => !b.includes(i)),
     };
 }
+
+/**
+ * Helper functions to perform LINQ-like queries on arrays.
```

Comment:
I see you're slowly implementing every C# / .NET Framework feature... 😜

Reply:
I *really* wanted `groupBy`